crates/remote/src/transport/ssh.rs

```diff
@@ -1544,8 +1544,14 @@ fn build_command(
     } else {
         write!(exec, "{ssh_shell} -l")?;
     };
+
+    // The command needs to be executed in a new shell, so it requires quoting again
+    let exec_after_quote = ssh_shell_kind
+        .try_quote(&exec)
+        .context("shell quoting")?
+        .into();
```

Comment:
~~This doesn't seem right, the shell builder should already be taking care of this~~ See my comment below